gammapy/stats/variability.py

```diff
@@ -228,3 +229,55 @@ def compute_flux_doubling(flux, flux_err, coords, axis=0):
     }
 
     return doubling_dict
+
+
+def structure_function(flux, flux_err, time, tdelta_precision=5):
+    """Compute the discrete structure function for a variable source.
+
+    Parameters
+    ----------
+    flux : `~astropy.units.Quantity`
+        The measured fluxes.
+    flux_err : `~astropy.units.Quantity`
+        The error on measured fluxes.
+    time : `~astropy.units.Quantity`
+        The time coordinates at which the fluxes are measured.
+    tdelta_precision : int, optional
```

Comment:
Is `int` the correct option here, or should it be in units of time.
Of course, this conversion can also be done while calling this function

Reply:
Thanks @AtreyeeS! Yes, it is quite easy to do the conversion during the call. I decided to pass an int as it is the direct way it is utilized in the code, and I don't think the parameter will be modified much often by the user.

Reply:
@cgalelli . Does it really work if you specify the unit? What do you gain with the rounding? Can't one use the full precision?

Reply:
This would not affect most cases in which the time bins are uniform, or most of the cases with very different inhomogeneous bins. Since the structure function is binned on the time distances between measurements, it can happen that in cases with inhomogeneous but similar spacing, you can have two sets of measures with distances, purely for example, 1.7845685 hours and 1.7845632 hours. The 'tdelta_precision' parameter I use is there to specify at which decimal to check if the distances are "the same". Maybe in some use cases a rough binning is fine, or for example it is necessary due to having few data points, so one can relax the check by lowering the precision. In other cases instead the user has plenty of measurements or needs precision.